### PR TITLE
Feature/search select

### PR DESCRIPTION
**Pull request by tnagorra**

### CODE REVIEW

File: src/components/SelectInput/EmptyOptions/index.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+function EmptyOptions(props: DefaultEmptyComponentProps) {
+    const {
+        isFiltered = false,
+        optionsPending = false,
+    } = props;
+
+    if (optionsPending) {
+        // FIXME: use loading
```
Comment (AdityaKhatri) [inline src/components/SelectInput/EmptyOptions/index.tsx:17]: Why not bring Loading from client

File: src/components/SelectInput/index.tsx [hunk 2]
```diff
@@ -1,143 +1,71 @@
...
-                props.onChange(undefined, name);
-            }
-        },
-        [name, props.onChange, props.nonClearable],
+    const searchOptions = React.useMemo(
+        () => rankedSearchOnList(options ?? (emptyList as O[]), searchInputValue, labelSelector),
+        [options, searchInputValue, labelSelector],
     );
 
+    // NOTE: this looks weird but we need to use typeguard to identify between
```
Comment (AdityaKhatri) [inline src/components/SelectInput/index.tsx:38]: I agree this looks weird

**Comment by AdityaKhatri** [comment]
For some reason, after de-selecting in SearchMultiSelect, the options popup closes.

File: src/components/SelectInput/EmptyOptions/index.tsx (hunk 1 above)
Comment (tnagorra) [reply src/components/SelectInput/EmptyOptions/index.tsx:17]: Because, i don't think it's ready yet.

**Comment by tnagorra** [comment]
> For some reason, after de-selecting in SearchMultiSelect, the options popup closes.

From my initial observation, looks like the problem is from useBlurEffect hook. @frozenhelium any idea why this is happening?

**Pull request merged**